pandas/core/computation/scope.py

```diff
@@ -223,6 +236,7 @@ def swapkey(self, old_key, new_key, new_value=None):
         maps.append(self.temps)
 
         for mapping in maps:
+            assert isinstance(mapping, (DeepChainMap, dict)), type(mapping)
```

Comment:
I'd prefer not to see this and have the dictionaries making up the DeepChainMap correctly recognised as a MutableMapping but I think that's out of scope here.

IIUC the maps of a normal ChainMap are immutable and writing to a ChainMap stores the new value in the top-level.